src/routes/transactions/mappers/common/native-staking.mapper.ts

```diff
@@ -257,6 +260,52 @@ export class NativeStakingMapper {
     });
   }
 
+  /**
+   * Dynamically returns the value to withdraw or withdrawn based on whether
+   * the batchWithdrawCLFee transaction was executed or not. This is done to
+   * cater for Kiln's API.
+   *
+   * Before execution the API returns current `net_claimable_consensus_rewards`
+   * and after execution it returns 0. Therefore, if the transaction was executed
+   * we return the value get the exact value from the transaction logs instead.
+   *
+   * @param {MultisigTransaction | ModuleTransaction | null} args.transaction - the `batchWithdrawCLFee` transaction
+   * @param {DataDecoded} args.dataDecoded - the decoded data of the transaction
+   * @param {string} args.chainId - the chain ID of the native staking deployment
+   *
+   * @returns {number} the value to withdraw or withdrawn
+   */
+  private async getWithdrawValue(args: {
+    transaction: MultisigTransaction | ModuleTransaction | null;
+    dataDecoded: DataDecoded;
+    chainId: string;
+    safeAddress: `0x${string}`;
+  }): Promise<number> {
+    if (!args.transaction?.transactionHash) {
+      return this.kilnNativeStakingHelper.getValueFromDataDecoded({
+        chainId: args.chainId,
+        safeAddress: args.safeAddress,
+        dataDecoded: args.dataDecoded,
+      });
+    }
+
+    const txStatus = await this.stakingRepository.getTransactionStatus({
+      chainId: args.chainId,
+      txHash: args.transaction.transactionHash,
+    });
+
+    const value = txStatus.receipt.logs
+      .map((log) => this.kilnDecoder.decodeWithdrawal(log))
+      .reduce((acc, cur) => {
+        if (cur === null) {
+          return acc;
+        }
+        return acc + cur.rewards;
+      }, BigInt(0));
+
+    return Number(value);
```

Comment:
Nit: At this point `value` is `BigInt`. It would be likely more correct to have `BigInt` also as the function return type, but I guess that requires an additional, potentially big, refactor. Wdyt? At some point it could be beneficial to rethink some types that we expect to hold big numbers.

Reply:
I agree that would make sence but the value returned by `getValueFromDataDecoded` is `number`. Do you think we should return `BigInt` from that instead?

Reply:
Yes, I think so, but I'd tackle this separately as it could be a major (and risky) refactor. 

In fact, it doesn't seem to make a lot of sense to do the math using `BigInt` and casting the result to `Number` after, the precision is lost anyway:

```javascript
> Number(BigInt(64 * 1e18) + BigInt(1))
64000000000000000000
```